### PR TITLE
Improve play/pause API, add playToggle method

### PR DESCRIPTION
**Pull request by johnBartos**
### This PR will...
- `play(true)` -> `play()`
- `play(false)` -> `pause()`
- `play(?null)` -> `playToggle()`
- `pause(true)` -> `pause()`
- `pause(false)` -> `play()`
- `pause(?null)` -> `playToggle()`

### Why is this Pull Request needed?
It makes the API more understandable

### Are there any points in the code the reviewer needs to double check?
Did I miss any instances of play/pause, or did I mistake previous usages of play/pause?

### Are there any Pull Requests open in other repos which need to be merged with this?
https://github.com/jwplayer/jwplayer-commercial/pull/4088

#### Addresses Issue(s):

JW8-541

### CODE REVIEW

File: src/js/view/controls/components/timeslider.js [hunk 1]
```diff
@@ -201,7 +201,7 @@ class TimeSlider extends Slider {
         var duration = this._model.get('duration');
         var position;
         if (duration === 0) {
-            this._api.play(reasonInteraction());
+            this._api.playToggle(reasonInteraction());
```
Comment (robwalch) [inline src/js/view/controls/components/timeslider.js]: This doesn't make sense. `play` was not being used as intended. See:

https://github.com/jwplayer/jwplayer/commit/fd47a76df76b94aba4350bec5df2a0fae7efea3d#diff-5b00ecc1e80bb1fce91c8111d655806eL137

and

https://github.com/jwplayer/jwplayer/commit/9b3985aa4fc3cf0110321522955c43ba2aa72fc6#diff-5b00ecc1e80bb1fce91c8111d655806eR88

I think the intent was to initiate playback (call `play(true)`) when the stream hasn't begun. We definitely do not want to call `video.pause()` here.

File: src/js/compatibility.js [hunk 2]
```diff
@@ -108,9 +109,55 @@
                         }
                     });
                 }
-                return setup.call(this, options);
+
+                return setup.call(basePlayer, options);
```
[the remark this answers is not in the document]
Comment (johnBartos) [reply src/js/compatibility.js]: You're right, we're not getting events. Maybe I can rebind trigger to the playerInstance.

[the remark this answers is not in the document]
Comment (robwalch) [reply src/js/compatibility.js]: Have a look at how `.on` runs and if it's using the wrong context.

File: src/js/compatibility.js [hunk 3]
```diff
@@ -18,23 +18,31 @@
...
 
             /*
                 We've removed a few methods from the public API, and our events now implement Backbone events.
              */
-            playerInstance.dispatchEvent = playerInstance.trigger;
-            playerInstance.removeEventListener = playerInstance.off;
-            playerInstance.getItem = playerInstance.getPlaylistIndex;
-            playerInstance.getMeta = playerInstance.getItemMeta;
+            playerInstance.dispatchEvent = basePlayer.trigger;
+            playerInstance.removeEventListener = basePlayer.off;
```
Comment (robwalch) [inline src/js/compatibility.js]: `playerInstance.removeEventListener` should also be set to  `playerInstance.off.bind(basePlayer)`

**Pull request merged**